addons/account/models/account_bank_statement.py

```diff
@@ -132,24 +132,28 @@ def _compute_starting_balance(self):
                 # Need default value
                 statement.balance_start = statement.balance_start or 0.0
 
-    @api.depends('previous_statement_id', 'previous_statement_id.balance_end_real')
+    @api.depends('previous_statement_id', 'previous_statement_id.balance_end')
```

Comment:
I think it should still be depending on `balance_end_real`
@csnauwaert right?

Also, shouldn't it also be depending on `line_ids.amount`?

Reply:
it should still depends on balance_end_real indeed. However I'm not sure there is a need to depends on lines_ids.amount as if you add a line the balance_end_real will be different from the balance_end (computed field) and you won't be able to post the statement, so you will have to modify one way or another the balance_end_real.

Reply:
From what I understand, depending on `previous_statement_id.balance_end_real` is what triggers the recursion.

Reply:
I pushed a commit adding an assertion that should be passing with the changes done here

Reply:
@william-andre Indeed, the new test doesn't pass with de modified dependency. But, again, IMO, keeping `previous_statement_id.balance_end_real` won't solve the recursion error. Do you think there is an alternative that does not involve modifying the dependencies of `_compute_ending_balance()`?

Reply:
I have tried playing a bit with  `env.protecting`, `env.remove_to_compute` and `env.norecompute`, without success. But I think we should be able to do something with that. As this seems to be a bit tricky, maybe ask help to the ORM team?